src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -645,6 +677,60 @@ protected override void CreateIndicesForNonTypeMembers(ITypeDefinition typeDef)
             implementingMethods.Free();
         }
 
+        private SymbolChange FixSymbolChangeForReAddedMembers(IDefinition item, SymbolChange change)
+        {
+            // If this is a field that is being added, but it's part of a property or event that has been deleted
+            // and is now being re-added, we don't want to add the field twice, so we ignore the change.
+            // Unlike properties and methods, since we can't replace a field with a MissingMethodException
+            // we don't need to update it at all.
```

Comment:
We'll want to update the field's name to "_Deleted" at some point (see [https://github.com/dotnet/runtime/issues/75154](https://github.com/dotnet/runtime/issues/75154))

Reply:
We don't support deleting fields at all yet, so the comment could definitely have a "for now" modifier.

I did see that issue though, and it's not clear to me whether we would want our deleted symbols here to have the name "_Deleted" or if we'd just write the name to the metadata table as "_Deleted", to ensure that we can still find the symbols in order to "undelete" them later.. or if thats just an implementation detail of Roslyn.. or if indeed that matters at all 🤷‍♂️

Reply:
I think that's impl detail. If we remember the token of the deleted member and can update it back to a new name when re-added/renamed back it doesn't matter much if the symbol has the _Deleted name or the metadata writer special cases deleted members.